vividus-extension-selenium/src/main/java/org/vividus/bdd/monitor/AbstractScreenshotOnFailureMonitor.java

```diff
@@ -98,10 +101,27 @@ public void onAssertionFailure(AssertionFailedEvent event)
 
     protected abstract Optional<Screenshot> takeAssertionFailureScreenshot(String screenshotName);
 
-    private static boolean takeScreenshotOnFailure(Method method)
+    private boolean takeScreenshotOnFailure(Method method)
     {
-        return method != null && (method.isAnnotationPresent(TakeScreenshotOnFailure.class) || method
-                .getDeclaringClass().isAnnotationPresent(TakeScreenshotOnFailure.class));
+        boolean isAnnotationPresent = method != null && (method.isAnnotationPresent(TakeScreenshotOnFailure.class)
+                                     || method.getDeclaringClass().isAnnotationPresent(TakeScreenshotOnFailure.class));
+        if (isAnnotationPresent)
+        {
+            TakeScreenshotOnFailure annotation = method.isAnnotationPresent(TakeScreenshotOnFailure.class)
+                                                 ? method.getAnnotation(TakeScreenshotOnFailure.class)
+                                                 : method.getDeclaringClass()
+                                                 .getAnnotation(TakeScreenshotOnFailure.class);
+            String debugModeProperty = annotation.onlyInDebugMode();
+            if (!debugModeProperty.isEmpty())
+            {
+                if (debugModes == null)
+                {
+                    return false;
+                }
+                return debugModes.stream().anyMatch(s -> debugModeProperty.contains(s));
```

Comment:
For using in annotation smth like @TakeScreenshotOnFailure(onlyInDebugMode = "proxy, ui, .., ")

Reply:
`onlyInDebugMode` should contain a single debug mode, if collection is needed, it will be implemented separately, and collection will be used as an annotation field